lib/rules/selector-id-pattern/index.cjs

```diff
@@ -38,20 +39,24 @@ const rule = (primary) => {
 				return;
 			}
 
-			parseSelector(ruleNode.selector, result, ruleNode)?.walkIds((selectorNode) => {
-				if (normalizedPattern.test(selectorNode.value)) {
+			parseSelector(getRuleSelector(ruleNode), result, ruleNode)?.walkIds((selectorNode) => {
+				const { value, sourceIndex: index } = selectorNode;
+
+				if (normalizedPattern.test(value)) {
 					return;
 				}
 
-				const selector = String(selectorNode);
+				const selector = String(selectorNode).trim();
```

Comment:
The warnings were incorrect for the following pattern.

```css
/* a comment */
#A, /* a comment */
#B,
/* a comment */
#C {
  color: red;
}
```

<img width="557" alt="image" src="https://github.com/user-attachments/assets/312ae6b6-92a7-43d0-a846-060406ddd9db">

Reply:
see https://github.com/stylelint/stylelint/pull/7959#discussion_r1730295653

i.e.

```suggestion
				const selector = selectorNode.toString().trim();
```

Reply:
This change should be made in the `.mjs` file and then the project needs to be built.